### PR TITLE
refactor: update rules, tools and dependencies

### PR DESCRIPTION
**Pull request by Lodin**
Update is necessary to fit the new TypeScript 5.5 version used in the Vaadin projects.

### Changes

- Updated dependencies,
- Removed deprecated and removed rules,
- Updated the latest rule check tools,
- Added new `tsconfig.recommended.json` which could be used as a reference for the projects, since TS continue introducing new options. Note though that settings have some specific rules that could break a lot of code, so consider disabling them during migration.

### NOTE
Since not every plugin is ready for eslint 9+, the current version remains 8+.

### CODE REVIEW

File: src/rules/eslint/possible-problems.ts [hunk 1]
```diff
@@ -1,78 +1,79 @@
...
-    ],
-    'no-use-before-define': ['error', { classes: true, functions: true, variables: true }],
-    'no-useless-backreference': 'error',
-    // appears to be too problematic to use
-    'require-atomic-updates': 'off',
-    'use-isnan': 'error',
-    'valid-typeof': ['error', { requireStringLiterals: false }],
-  },
+    rules: {
+        "array-callback-return": "error",
```
Comment (web-padawan) [inline src/rules/eslint/possible-problems.ts:3]: Should we apply Prettier formatting to this file? Especially single quotes and 2 tab width.

File: src/rules/typescript-requiring-type-checking/extensions.ts [hunk 2]
```diff
@@ -1,21 +1,33 @@
-const suggestions = require('../eslint/suggestions').rules;
+const suggestions = require("../eslint/suggestions").rules;
```
Comment (web-padawan) [inline src/rules/typescript-requiring-type-checking/extensions.ts:1]: Should we apply Prettier formatting to this file too?

**Review by web-padawan — COMMENTED** [review]
Looks like `.ts` files under `src/rules` need to be formatted with Prettier.

File: src/rules/eslint/possible-problems.ts (hunk 1 above)
Comment (Lodin) [reply src/rules/eslint/possible-problems.ts:3]: I wonder why it's not applied automatically... Ok, will do!

**Review by web-padawan — APPROVED** [review]

**Pull request merged**